### ISSUE TITLE
Debugger does not respect timer

### ISSUE THREAD

**Issue by hankduan** [joyent/node#9333]
This was originally https://github.com/joyent/node/issues/9319, but I've reduced the case down more.
Note: this is happening with node 0.10.31 and above.

Given code `example.js`:

```javascript
setInterval(function() {  
  console.log(1)
  debugger
  console.log(2);
}, 10);
```

ok I enter into debugger now:

```
$ node debug ./example.js 
< Debugger listening on port 5858
connecting to port 5858... ok
break in example.js:1
> 1 setInterval(function() {  
  2   console.log(1)
  3   debugger
debug> c
< 1
break in example.js:3
  1 setInterval(function() {  
  2   console.log(1)
> 3   debugger
  4   console.log(2);
  5 }, 10);
debug> c    //*********now wait 1min before you hit enter******//
< 2 //********now it gets stuck here for 1min before it moves onto the next break point****//
```

My assumption is that the break points are messing up node's timer some how.

This may seem like a small issue in the debugger, but it's causing huge pains for Protractor users because all of protractor's code (which is using webdriver) is in a giant setInterval.

reference: 
https://github.com/angular/protractor/issues/1890
https://github.com/angular/protractor/issues/1899
https://github.com/angular/protractor/issues/1822
https://github.com/angular/protractor/issues/1890
https://github.com/angular/protractor/issues/1893

and the only workaround for that right now is to downgrade node to 0.10.30 or lower. 

Thanks!

**Comment by hankduan** [comment]
fyi: the break is because of this change: https://github.com/joyent/node/commit/6f043940bdcbfb5272be8ae959cd74b9fb5cf4f8#diff-a7a2d51d79b1c80c7458e0d056e3d86fR167

i.e. if I remove this line (https://github.com/joyent/node/blob/v0.12.0/src/timer_wrap.cc#L150) it is fixed. However, I don't know enough about node to make the fix without breaking that commit's intentions unfortunately =(

**Comment by jperry** [comment]
please fix asap as this is a painful bug.

**Comment by misterdjules** [comment]
Closing as duplicate of #9125, but the information in this issue may be very helpful to investigate it, so thank you very much @hankduan! :+1:

**Comment by misterdjules** [comment]
Actually, after investigating this more carefully, it's not clear to me it's a duplicate of #9125. Reopening.

**Comment by misterdjules** [comment]
It seems it also happens on v0.12.

**Comment by misterdjules** [comment]
This is the same issue as #15447. The [diff proposed there](https://github.com/joyent/node/issues/15447#issue-68605357), although it will not necessarily be committed as is, fixes this issue.

**Comment by misterdjules** [comment]
Created a PR that fixes the issue: #17203. Thank you very much @hankduan!

**Comment by hankduan** [comment]
Thanks a bunch!

**Comment by misterdjules** [comment]
Fixed by https://github.com/joyent/node/commit/d38e865fce93c00cb128034202fb1e26987efdbc, which will be available in the upcoming v0.10.39 and v0.12.5 releases.

**Comment by hankduan** [comment]
Thanks @misterdjules!
